unit_of_time/__init__.py

```
@@ -165,6 +186,12 @@ def to_str(cls, dt):
     def truncate(cls, dt):
         return date(dt.year, 3 * ((dt.month - 1) // 3) + 1, 1)
 
+    def _inner_shift(cls, cur, dt, amount):
+        q_new = dt.year * 4 + amount + (dt.month - 1) // 3
+        y = q_new // 4
```

Ruff (0.12.2)

Local variable `y` is assigned to but never used

Remove assignment to unused variable `y`

(F841)